### PR TITLE
[WIP] build: add poetry build configuration

### PR DESCRIPTION
**Pull request by afuetterer**
## Proposed changes

This PR proposed a [poetry](https://python-poetry.org/) based build. Let's see if the GitHub actions run through.
In my experience poetry is faster and offers more features than pipenv.

### CODE REVIEW

**Comment by afuetterer** [comment]
Is the pyproject.toml not found during install?

**Comment by hansendx** [comment]
Thank you for the pull request.
I'm open to changing towards poetry but I'm going to need to look further into it's workflow first.

The pyproject.toml is not inside the container during tests runs.
Dependency and config files are individually mounted in the [docker-compose-ci.yml](https://github.com/ddionrails/ddionrails/blob/master/docker-compose-ci.yml).
So the compose file needs to be changed here as well.

**Comment by afuetterer** [comment]
I see, let me fix that then.

File: .github/workflows/system-tests.yml [hunk 1]
```diff
@@ -32,7 +32,10 @@ jobs:
           docker images
           docker ps -a
           docker exec ddionrails_web_1 bash -c \
-          'pipenv requirements --dev > Requirements.txt'
+          'pip install poetry==1.3.2'
+          docker exec ddionrails_web_1 bash -c \
+          'poetry export --output=Requirements.txt'
+          docker exec ddionrails_web_1 pip uninstall -y djangorestframework
```
Comment (afuetterer) [inline .github/workflows/system-tests.yml:38]: fixed in develop?

**Pull request closed without merging**